src/assets.zig

```diff
@@ -25,15 +43,25 @@ pub fn readAllZonFilesInAddons(externalAllocator: NeverFailingAllocator, addons:
 			continue;
 		};
 		defer dir.close();
-
+		
+		var defaultMap = std.StringHashMap(ZonElement).init(main.stackAllocator.allocator);
+		defer {
+			var iter = defaultMap.iterator();
+			while (iter.next()) |entry| {
+				main.stackAllocator.free(entry.key_ptr.*);
+				entry.value_ptr.*.deinit(main.stackAllocator);
+			}
+			defaultMap.deinit();
+		}
```

Comment:
To avoid this lengthy cleanup here, you could just use an ArenaAllocator for everything that goes into the map.